bloggies_backend/src/routes/stripe.ts

```diff
@@ -19,45 +20,72 @@ stripeRouter.post("/webhook", async function (req: Request, res: Response, next:
   let event = req.body;
   let data: any;
   let sub: Stripe.Subscription;
-  try {
+
+  try {    
     switch (event.type) {
       case 'invoice.upcoming':
-        // invoice object
+        //invoice object
         data = event.data.object;
-        console.log(`invoice upcoming, subscription almost ending for  cust ${data.customer}`);
-        break;
+        sub = await stripe.subscriptions.retrieve(data.subscription);
+        const user = await User.getUserBySubscriptionId(sub.id);
+
+        //check today against cancel_at date to see if user is overdue for a post
+        const now = new Date();
+        const isOverdue = now >= user.cancel_at;
+        const unixNow = Math.floor(now.getTime() / 1000);
+
+        if(isOverdue) {
+          try{
+            await Checkout.stripeSubscriptionCancel(user.user_id);
+            await User.cancelSubscription(user.user_id, unixNow);
```

Comment:
Is there a reason you are putting the cancelSubscription in the invoice.upcoming instead of letting the customer.subscription.deleted case handle updating the database?

Reply:
I have the same question too!

Reply:
Oh, no! That was negligence. I will change that. I did the same thing in the login so I will change that as well.

Reply:
Fixed!